pulp_ansible/app/serializers.py

```diff
@@ -115,47 +116,6 @@ class Meta:
         model = CollectionRemote
 
 
-class CollectionOneShotSerializer(serializers.Serializer):
-    """
-    A serializer for the Collection One Shot Upload API.
-    """
-
-    file = serializers.FileField(help_text=_("The Collection tarball."), required=True)
-
-    sha256 = serializers.CharField(
-        help_text=_("An optional sha256 checksum of the uploaded file."),
-        required=False,
-        default=None,
-    )
-
-    expected_namespace = serializers.CharField(
```

Comment:
We need the user to be able to submit these 3 attributes as the "expected names" to be validated against the data returned from `importer_result = import_collection(file=artifact_file)` line in `deferred_validate()`. To do that we probably still need pulp_ansible to use it's own viewset and serializer that extends the base provided ones.

Reply:
It should be sufficient, to include them as write_only. Then they will be available in `deferred_validation`.